cmd/clusterctl/examples/vsphere/machineset.yaml.template

```diff
@@ -1,40 +1,36 @@
-apiVersion: "cluster.k8s.io/v1alpha1"
+apiVersion: cluster.k8s.io/v1alpha1
 kind: MachineSet
 metadata:
-  name: ${CLUSTER_NAME}-machineset
+  name: "${CLUSTER_NAME}-machineset-1"
 spec:
   replicas: 2
   selector:
     matchLabels:
-      node-type: worker-node
-      cluster.k8s.io/cluster-name: ${CLUSTER_NAME}
+      machineset-name: "${CLUSTER_NAME}-machineset-1"
+      cluster.k8s.io/cluster-name: "${CLUSTER_NAME}"
   template:
     metadata:
       labels:
-        node-type: worker-node
-        cluster.k8s.io/cluster-name: ${CLUSTER_NAME}
+        machineset-name: "${CLUSTER_NAME}-machineset-1"
+        cluster.k8s.io/cluster-name: "${CLUSTER_NAME}"
     spec:
       providerSpec:
         value:
-          apiVersion: "vsphereproviderconfig/v1alpha1"
-          kind: "VsphereMachineProviderConfig"
+          apiVersion: vsphereproviderconfig/v1alpha1
+          kind: VsphereMachineProviderConfig
           machineSpec:
             datacenter: "${VSPHERE_DATACENTER}"
             datastore: "${VSPHERE_DATASTORE}"
             resourcePool: "${VSPHERE_RESOURCE_POOL}"
             vmFolder: "${VSPHERE_FOLDER}"
-            networks:
-            - networkName: "${VSPHERE_NETWORK}"
-              ipConfig:
-                networkType: "dhcp"
-            numCPUs: 2
-            memoryMB: 2048
+            network:
+              devices:
+              - networkName: "${VSPHERE_NETWORK}"
+                dhcp4: true
+                dhcp6: true
+            numCPUs: "${VSPHERE_NUM_CPUS}"
+            memoryMB: "${VSPHERE_MEM_MIB}"
+            diskGiB: "${VSPHERE_DISK_GIB}"
```

Comment:
This shouldn't be quoted since `diskGiB` is an `int`
```
E0624 19:40:58.293182       1 machine_controller.go:202] Failed to check if machine "capv-mgmt-example-controlplane-1" exists: failed to load machine provider config: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go struct field VsphereMachineSpec.diskGiB of type int32
``